lib/blocks.php

```diff
@@ -296,3 +296,209 @@ function gutenberg_enqueue_editor_block_styles_assets() {
+		$has_custom_text_color = isset( $attributes['style']['color']['text'] );
+
+		// Apply required generic class.
+		if ( $has_custom_text_color || $has_named_text_color ) {
+			$color_settings['css_classes'][] = 'has-text-color';
+		}
+		// Apply color class or inline style.
+		if ( $has_named_text_color ) {
+			$color_settings['css_classes'][] = sprintf( 'has-%s-color', $attributes['textColor'] );
+		} elseif ( $has_custom_text_color ) {
+			$color_settings['inline_styles'] .= sprintf( 'color: %s;', $attributes['style']['color']['text'] );
+		}
+	}
+
+	// Link Colors.
+	if ( in_array( 'link-color', $supports, true ) ) {
+		$has_link_color = isset( $attributes['style']['color']['link'] );
+		// Apply required class and style.
+		if ( $has_link_color ) {
+			$color_settings['css_classes'][] = 'has-link-color';
+			// If link is a named color.
+			if ( strpos( $attributes['style']['color']['link'], 'var:preset|color|' ) !== false ) {
+				// Get the name from the string and add proper styles.
+				$index_to_splice                  = strrpos( $attributes['style']['color']['link'], '|' ) + 1;
+				$link_color_name                  = substr( $attributes['style']['color']['link'], $index_to_splice );
+				$color_settings['inline_styles'] .= sprintf( '--wp--style--color--link:var(--wp--preset--color--%s);', $link_color_name );
+			} else {
+				$color_settings['inline_styles'] .= sprintf( '--wp--style--color--link: %s;', $attributes['style']['color']['link'] );
+			}
+		}
+	}
+
+	// Background Colors.
+	if ( in_array( 'background-color', $supports, true ) ) {
+		$has_named_background_color  = array_key_exists( 'backgroundColor', $attributes );
+		$has_custom_background_color = isset( $attributes['style']['color']['background'] );
+
+		// Apply required background class.
+		if ( $has_custom_background_color || $has_named_background_color ) {
+			$color_settings['css_classes'][] = 'has-background';
+		}
+		// Apply background color classes or styles.
+		if ( $has_named_background_color ) {
+			$color_settings['css_classes'][] = sprintf( 'has-%s-background-color', $attributes['backgroundColor'] );
+		} elseif ( $has_custom_background_color ) {
+			$color_settings['inline_styles'] .= sprintf( 'background-color: %s;', $attributes['style']['color']['background'] );
+		}
+	}
+
+	// Gradients.
+	if ( in_array( 'background', $supports, true ) ) {
+		$has_named_gradient  = array_key_exists( 'gradient', $attributes );
+		$has_custom_gradient = isset( $attributes['style']['color']['gradient'] );
+
+		if ( $has_named_gradient || $has_custom_gradient ) {
+			$color_settings['css_classes'][] = 'has-background';
+		}
+		// Apply required background class.
+		if ( $has_named_gradient ) {
+			$color_settings['css_classes'][] = sprintf( 'has-%s-gradient-background', $attributes['gradient'] );
+		} elseif ( $has_custom_gradient ) {
+			$color_settings['inline_styles'] .= sprintf( 'background: %s;', $attributes['style']['color']['gradient'] );
+		}
+	}
```

Comment:
The `color`, `background-color`, and `background` clauses seem quite similar. Would it be worth extracting a helper function?

Reply:
Maybe?  Although it seems that helper function would require 6+ specific params in each case.  Maybe its best to leave them separate in case they need individual customization in the future?

Reply:
Ugh, 6? Granted, I didn't count. Yeah, that might not make too much sense then.

Reply:
If im thinking about it right:  the support key, the attribute name, the style attribute name, the required class to apply, the named style classname pattern, and the inline style pattern.